packages/js/product-editor/src/components/iframe-editor/iframe-editor.tsx

```diff
@@ -42,20 +43,28 @@ type IframeEditorProps = {
 };
 
 export function IframeEditor( {
-	closeModal = () => {},
 	initialBlocks = [],
 	onChange = () => {},
 	onClose,
 	onInput = () => {},
 	settings: __settings,
 }: IframeEditorProps ) {
 	const [ resizeObserver ] = useResizeObserver();
-	const [ blocks, setBlocks ] = useState< BlockInstance[] >( initialBlocks );
 	const [ temporalBlocks, setTemporalBlocks ] =
 		useState< BlockInstance[] >( initialBlocks );
+
+	// Pick the blocks from the store.
+	const blocks: BlockInstance[] = useSelect( ( select ) => {
+		return select( productEditorUiStore ).getModalEditorBlocks();
+	}, [] );
+
+	const { setModalEditorBlocks: setBlocks, setModalEditorContentHasChanged } =
+		useDispatch( productEditorUiStore );
```

Comment:
Is there a specific reason you still kept this as part of the iframe editor?
I was kinda thinking we could trigger these functions from the `onChange` menu in the `ModalEditor` component.

Likewise for the `blocks` I was thinking we could use the `initialBlocks`, or just pass in the `blocks` array so there is no intermediary state.